scripts/SANS/sans/state/StateObjects/StateMoveDetectors.py

```diff
@@ -276,6 +276,13 @@ def __init__(self, data_info):
 
     def _set_conversion_value(self, data_info):
         run_number = data_info.sample_scatter_run_number
+        if not run_number:
+            # We are going through init before file was selected, lets assume > 2217 so
+            # the division is the same as Pos 2 in the case they load a later file
+            # when a file gets loaded this re-runs anyway overwriting our assumptions
+            run_number = 2220
```

Comment:
Are we strictly happy to use a hard-coded value here? Couldn't we just say:
```python
if run_number == None or run_number > 2217:
    self.conversion_value = 1.
```

Reply:
@DavidFair This is probably an indication that we need to think a bit more about how units are handles so we can shift the cognitive burden off the developer or this will become difficult to develop into.

Reply:
Agree on folding the if block into a single condition. I was expecting us to have to do something fancy, but it turns out to just be they literally enter one unit as meters and one as mm so I didn't press on how we ended up here

I was saying to Gemma yesterday some sort of set of decorators available to all GUIs which converts from user friendly values into SI units internally could work such as?

```
GuiModel(object):
   def __init__(self):
        self._my_val = 0.012  # In meters

   @property
   @convert_to(units.mm)
   def my_val(self)
       return self._my_val

   @my_val.setter
   @convert_to(units.mm)
   def my_val(self, value)
       return self._my_val
```

Then you would have something like
```
m = GuiModel()
print(m.my_val)  # Returns 0.012 as its internal
print(m.my_val_view)  # <name>_view is generated and correctly applies a scale value to display 12mm

m.my_value_view = view.get_my_value()  # Internally scales the users mm meters then calls the property setter
```


Python doesn't give us much in terms of typing tools without throwing a layer of C++ on top so I imagine this will remove the cognitive burden enough, as exposure to view code tends to be limited

Reply:
I think unit handling goes beyond just this interface and is perhaps something which needs more careful thought.

Reply:
I do like the idea of further providing more GUI building blocks though, much like `FileFinderWidget` and if we do a lot of entry where numbers are entered in one unit and convert to another then a widget for handling that is useful for others to build with.

There is additional complication here as the values are dependent on what run number is entered? How do we deal with that in a sane/general manner. Should we? I think there are times when the right answer might be to say we can't deal with everything that might be wrong with input data.
We do have a mechanism for dealing with changing length of [run numbers](https://github.com/mantidproject/mantid/blob/master/instrument/Facilities.xml#L31) but we probably want to be careful at what level we go to allow complete freedom to "fix" bad data.

Reply:
In this case I would advocate for simplicity rather than edge cases
This problem will go away for SANS too with the migration to TOML files, which partly was to resolve historical weirdness like this

Having some widget which correctly labelled, converted and displayed to a custom precision (to avoid FP rounding errors) would likely get significant usage.